multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/v2/ResourceLiveParameterResolver.java

```diff
@@ -0,0 +1,149 @@
+    private DeploymentDescriptor deploymentDescriptor;
+
+    public ResourceLiveParameterResolver(DeploymentDescriptor deploymentDescriptor) {
+        this.deploymentDescriptor = deploymentDescriptor;
+    }
+
+    public void resolveResource(Resource resource) {
+        if (resource.getParameters()
+                    .containsKey(CONFIG)) {
+            Map<String, Object> serviceCreationParameters = castToMap(resource.getParameters()
+                                                                              .get(CONFIG));
+            serviceCreationParameters.entrySet()
+                                     .forEach(entry -> resolveResourceParameter(entry, resource));
+        }
+    }
+
+    private Map<String, Object> castToMap(Object entryValue) {
+        return (Map<String, Object>) entryValue;
+    }
+
+    private void resolveResourceParameter(Map.Entry<String, Object> entry, Resource resource) {
+        if (entry.getValue() instanceof Map) {
+            resolveMapParameter(castToMap(entry.getValue()), resource);
+        } else if (entry.getValue() instanceof String) {
+            resolveConfigParameter(entry, resource);
+        }
+    }
+
+    private void resolveMapParameter(Map<String, Object> resourceParameter, Resource resource) {
+        resourceParameter.entrySet()
+                         .forEach(entry -> resolveResourceParameter(entry, resource));
+    }
+
+    private void resolveConfigParameter(Map.Entry<String, Object> resourceParameter, Resource resource) {
+        String resourceParameterValue = (String) resourceParameter.getValue();
+        List<Reference> matchedReferences = ReferencePattern.FULLY_QUALIFIED.match(resourceParameterValue);
+
+        matchedReferences.forEach(ref -> updateParamsFromReference(resource, ref, resourceParameterValue, resourceParameter));
+    }
+
+    private void updateParamsFromReference(Resource resource, Reference ref, String resourceParameterValue,
+                                           Map.Entry<String, Object> resourceParameter) {
+        deploymentDescriptor.getModules()
+                            .stream()
+                            .filter(module -> isModuleNeeded(module, ref.getDependencyName()))
+                            .findFirst()
+                            .ifPresent(module -> updateMapFromModule(resource, ref, module, resourceParameterValue, resourceParameter));
+    }
+
+    private void updateMapFromModule(Resource resource, Reference ref, Module module, String resourceParameterValue,
+                                     Map.Entry<String, Object> resourceParameter) {
+        Optional<ProvidedDependency> moduleProvidedDependencyOpt = module.getProvidedDependencies()
+                                                                         .stream()
+                                                                         .filter(dp -> dp.getName()
```

Comment:
avoid variable shortening like 'dp' and 'ref' since the full names themselves aren't that long anyways